### PR TITLE
Add contextual help on each page

### PR DESCRIPTION
**Pull request by haoweiqiu**
## Related issues 

#206 

## Checklist

- [x] **DO** Unit tests pass.
- [ ] **DO** If new feature, created unit test.

## Purpose

Resolve issue #206.

## Current behaviour

 There is no help text provided by the portal.

## New behaviour

Help texts are provided.

#### Does this introduce a major change?
- [ ] Yes
- [x] No

## Implementation Detail

page5
page1
page2
page3
page4

### CODE REVIEW

File: app/templates/common/base_main.html [hunk 1]
```diff
@@ -115,6 +115,143 @@ <h6>The user agrees not to:</h6>
...
+            <p>Note that some datasets may require a third-party account. Instructions on how to 
+            obtain an account for those datasets will be displayed on the dataset page if that is a requirement.</p>
+            <p>Below are two dataset examples, one that requires a third-party account (PREVENT-AD 
+            open data) and one that does not require any account to download it (BigBrain dataset).</p>
+            <img src="/static/img/Dataset_accounts.png"alt="Portal_access_from_conp.ca" width="700px">
+            <h5>How to download the data?</h5>
+            <p>Dataset download is available through DataLad. Each dataset page contains a detailed 
+            ‘Dataset Download Instructions’ section that lay out the different steps to perform in order 
+            to download the dataset.</p>
+            <img src="/static/img/Dataset_download_instructions.png"alt="Portal_access_from_conp.ca" width="700px">
```
Comment (cmadjar) [inline app/templates/common/base_main.html:174]: This part should not be in the data search page help section. I would instead modify it a little as follows and have that shown in the dataset specific page instead.

```suggestion
            <h4>Dataset Page</h4>
            <p>This page provides more detailed information regarding a dataset. This includes contact information, dataset version, descriptions, licenses, related publication and so on.</p>
            <p>Dataset download is available through DataLad. Each dataset page contains a detailed 
            ‘Dataset Download Instructions’ section that lay out the different steps to perform in order 
            to download the dataset.</p>
            <img src="/static/img/Dataset_download_instructions.png"alt="Portal_access_from_conp.ca" width="700px">
            <p>Finally, the dataset README file can be viewed at the end of the page.</p>
```

File: app/templates/common/base_main.html [hunk 2]
```diff
@@ -115,6 +115,143 @@ <h6>The user agrees not to:</h6>
...
+        <ul>
+            <h4>How can I find and run tools/pipelines?</h4>
+            <h5>How to find a tool/pipeline?</h5>
+            <p>The Tools & Pipelines page lists all softwares registered in the portal. Each tool 
+            or pipeline present in that list page contains basic information about the tool/pipeline 
+            and its pipeline ID.</p>
+            <img src="/static/img/Pipeline_search_page.png"alt="Portal_access_from_conp.ca" width="700px">
+            <p>Clicking on a specific choice will bring the user to a more detailed page with links 
+            to instructions on how to run the tool locally, to HCPs or to external platforms.</p>
+            <img src="/static/img/Pipeline_detailed_page.png"alt="Portal_access_from_conp.ca" width="700px">
```
Comment (cmadjar) [inline app/templates/common/base_main.html:201]: Same general comment as for the dataset section. This should be divided in two: one tools & pipelines search to be displayed on this page and one specific to the pipeline page to be displayed on the pipeline page.

```suggestion
            <h4>Tools & Pipelines Search Page</h4>
            <p>The Tools & Pipelines Search page lists all softwares registered in the portal. Each tool 
            or pipeline present in that list page contains basic information about the tool/pipeline 
            and its pipeline ID.</p>
            <img src="/static/img/Pipeline_search_page.png"alt="Portal_access_from_conp.ca" width="700px">
            <p>Clicking on a specific choice will bring the user to a more detailed page with links 
            to instructions on how to run the tool locally, to HCPs or to external platforms.</p>
            <img src="/static/img/Pipeline_detailed_page.png"alt="Portal_access_from_conp.ca" width="700px">
```

File: app/templates/common/base_main.html [hunk 3]
```diff
@@ -115,6 +115,143 @@ <h6>The user agrees not to:</h6>
...
+            <img src="/static/img/Boutiques_instructions.png"alt="Portal_access_from_conp.ca" width="700px">
+            <p>If you have never used Boutiques in the past, we recommend the following tutorial.</p>
+            <h6>Run the tool on HPCs</h6>
+            <p>Some tools and pipelines can also be run on HPCs via the CBRAIN infrastructure or in 
+            your Compute Canada account with Clowdr.</p>
+            <img src="/static/img/CBRAIN_link_highlighted.png"alt="Portal_access_from_conp.ca" width="700px">
+            <p>The 'Click here to begin' link under the 'Run on the CBRAIN Plateform' container will open a 
+            tab to the login page of CBRAIN or to your CBRAIN account if you are already logged in.</p>
+            <p>The 'Click here for instructions' link under the 'Run on HPCs with Clowdr' will open a 
+            new tab with the instructions to follow to deploy a tool/pipeline on your Compute Canada account.</p>
```
Comment (cmadjar) [inline app/templates/common/base_main.html:218]: This part should go into the pipeline specific page.

```suggestion
            <h4>Tool/Pipeline Page</h4>
            <p>The tool/pipeline detailed page displays all the information pertinent to the tool/pipeline. This includes authorship information, the descriptor URL, the container image information, the tool description and so on. In addition, the page provides information on how to run the tool locally or on HPCs via the CBRAIN platform or Clowdr.</p>
            <h6>Run the tool locally</h6>
            <p>Tools and pipelines can be run locally on any system using Boutiques and a container engine 
            (Docker or Singularity) based on the Zenodo ID of the tool. The container image for the tool is 
            listed in the detailed page of the tool/pipeline.</p>
            <p>At the bottom of the pipeline page, detailed instructions on how to run the tool locally 
            are available to the user.</p>
            <img src="/static/img/Boutiques_instructions.png"alt="Portal_access_from_conp.ca" width="700px">
            <p>If you have never used Boutiques in the past, we recommend the following tutorial.</p>
            <h6>Run the tool on HPCs</h6>
            <p>Some tools and pipelines can also be run on HPCs via the CBRAIN infrastructure or in 
            your Compute Canada account with Clowdr.</p>
            <img src="/static/img/CBRAIN_link_highlighted.png"alt="Portal_access_from_conp.ca" width="700px">
            <p>The 'Click here to begin' link under the 'Run on the CBRAIN Platform' container will open a 
            tab to the login page of CBRAIN or to your CBRAIN account if you are already logged in.</p>
            <p>The 'Click here for instructions' link under the 'Run on HPCs with Clowdr' will open a 
            new tab with the instructions to follow to deploy a tool/pipeline on your Compute Canada account.</p>
```

File: app/templates/common/base_main.html [hunk 4]
```diff
@@ -115,6 +115,143 @@ <h6>The user agrees not to:</h6>
...
+        <p>A dataset can be uploaded to CONP via the following mechanisms:</p>
+        <li>Through Zenodo by tagging your dataset with the keyword canadian-open-neuroscience-platform</li>
+        <li>Through the Open Science Framework (OSF) by tagging your dataset with the keyword canadian-open-neuroscience-platform
+         and setting the dataset to Public</li>
+         <li>Through DataLad within the context of CONP</li>
+         <p>Detailed technical instructions are available for all these options on the Share page of the portal.</p>
+         <h4>How can I upload my tool/pipeline?</h4>
+         <p>You can publish your tools/pipelines using Boutiques’ command bosh publish.</p>
+         <p>For more information on how to do that, please visit the ‘Publishing your own tool’ section of the 
+         Boutiques tutorial Python notebook.</p>
```
Comment (cmadjar) [inline app/templates/common/base_main.html:246]:
```suggestion
        <h4>Share Page</h4>
        <h5>How to Share Datasets and Pipelines Tab</h5>
        <p>This tab contains detailed technical instructions on how to upload your dataset or pipeline to CONP.</p> 
        <h6>Tool/Pipeline Upload</h6>
        <p>You can publish your tools/pipelines using Boutiques’ command bosh publish.</p>
        <p>For more information on how to do that, please visit the ‘Publishing your own tool’ section of the 
         Boutiques tutorial Python notebook.</p>
         <h6>Dataset Upload</h6>
        <p>A dataset can be uploaded to CONP via the following mechanisms:</p>
        <li>Through Zenodo by tagging your dataset with the keyword canadian-open-neuroscience-platform</li>
        <li>Through the Open Science Framework (OSF) by tagging your dataset with the keyword canadian-open-neuroscience-platform
         and setting the dataset to Public</li>
         <li>Through DataLad within the context of CONP</li>
         <h5>Create or Edit a DATS.json File Tab</h5>
         <p>Each dataset to be uploaded should contain a valid DATS.json descriptors derived from schema.org.
         We provide a DATS GUI editor that will create a DATS.json file to ease the creation process of the 
         DATS.json that will describe the dataset to be uploaded</p>
```

At the bottom of this help text, could you create a screenshot of the DATS GUI editor and show it after the short description of the editor?

**Review by cmadjar — CHANGES_REQUESTED** [review]
Thank you @haoweiqiu ! This is very nice!!

I suggested in the attached review some changes to the text of the help section. A few generic comments:
- Regarding the screenshots, did you take the images from the following [conp-documentation repo](https://github.com/CONP-PCNO/conp-documentation/tree/master/img)? Just to be sure they are the same. It looks like it is.
- in the edits I made, the image `app/static/img/Portal_access_from_conp.ca.png ` won't be displayed so it should be removed.
- split the help section of the search list and item pages into two for pipelines and datasets (more information provided in the inline comments)

Let me know once ready for another round of review. Thanks!

**Comment by haoweiqiu** [comment]
Hi @cmadjar, thank you very much for the comment. Yes, I took the images from the conp-documentation repo and I put the images under app/static since flask appears to consider this directory as default. I'll review your comment and modify soon.

File: app/templates/common/base_main.html [hunk 5]
```diff
@@ -115,6 +115,192 @@ <h6>The user agrees not to:</h6>
...
+            <p>Clicking on a dataset row in the Data search page will bring the user to a more 
+            detailed dataset page that includes more description information pertinent to the 
+            dataset as well as download instructions to follow using DataLad.</p>
+            <img src="/static/img/Dataset_detailed_page.png"alt="Dataset_detailed_page" width="700px">
+            <p>Note that some datasets may require a third-party account. Instructions on how to 
+            obtain an account for those datasets will be displayed on the dataset page if that is a requirement.</p>
+            <p>Below are two dataset examples, one that requires a third-party account (PREVENT-AD 
+            open data) and one that does not require any account to download it (BigBrain dataset).</p>
+            <img src="/static/img/Dataset_accounts.png"alt="Dataset_accounts" width="700px">
+            <img src="/static/img/Dataset_download_instructions.png"alt="Dataset_download_instructions" width="700px">
```
Comment (cmadjar) [inline app/templates/common/base_main.html:169]:
```suggestion
```

Comment (cmadjar) [reply app/templates/common/base_main.html:169]: remove this since it is specific to the dataset page, not the search page

**Review by cmadjar — CHANGES_REQUESTED** [review]
@haoweiqiu Thank you for the quick updates!

One thing to remove in the text content (see comment below).

Otherwise: one generic comment: is there a way to display the help content so that it does not block access to the portal page? So you could have the help open and keep scrolling and clicking on the page.

Thanks!

**Review by cmadjar — APPROVED** [review]
@liamocn also reviewed it and all good! Thank you @haoweiqiu :)

**Comment by haoweiqiu** [comment]
@cmadjar Thank you very much!

**Pull request merged**